contrib/http_base/src/test/java/io/opentelemetry/contrib/http/core/HttpClientHandlerTest.java

```diff
@@ -0,0 +1,193 @@
+import io.opentelemetry.sdk.trace.Samplers;
+import io.opentelemetry.sdk.trace.SpanData;
+import io.opentelemetry.sdk.trace.TracerSdkFactory;
+import io.opentelemetry.sdk.trace.config.TraceConfig;
+import io.opentelemetry.trace.Span;
+import io.opentelemetry.trace.Span.Kind;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.BeforeClass;
+import org.junit.Test;
+
+/** Unit tests for {@link HttpClientHandler}. */
+public class HttpClientHandlerTest {
+
+  private static final Logger LOGGER = Logger.getLogger(HttpClientHandlerTest.class.getName());
+  private static InMemoryTracing inMemoryTracing;
+
+  @BeforeClass
+  public static void configureTracing() {
+    TraceConfig traceConfig =
+        TraceConfig.getDefault().toBuilder().setSampler(Samplers.alwaysOn()).build();
+    TracerSdkFactory tracerSdk = (TracerSdkFactory) OpenTelemetry.getTracerFactory();
+    LOGGER.log(Level.INFO, "Original trace config: " + tracerSdk.getActiveTraceConfig());
+    tracerSdk.updateActiveTraceConfig(traceConfig);
+    LOGGER.log(Level.INFO, "Updated trace config: " + tracerSdk.getActiveTraceConfig());
+    inMemoryTracing = new InMemoryTracing(tracerSdk);
+  }
+
+  @Before
+  public void reset() {
+    if (inMemoryTracing != null) {
+      inMemoryTracing.reset();
+    }
+  }
+
+  @After
+  public void printSpans() {
+    for (SpanData span : inMemoryTracing.getFinishedSpanItems()) {
+      LOGGER.log(Level.FINE, span.toString());
+    }
+  }
+
+  @Test
+  public void shouldStartAndEndSpanWhenParentIsAvailable() {
+    Map<String, String> data = new HashMap<>();
+    HttpClientHandler<Map<String, String>, Map<String, String>, Map<String, String>> handler =
+        new HttpClientHandler<>(new TestOnlyMapHttpExtractor(), new TestOnlyMapGetterSetter());
+    Span parentSpan = constructParentSpan();
+    DistributedContext parentContext = constructParentContext();
+    LOGGER.log(Level.INFO, parentSpan.getContext().toString());
+    try (Scope parent =
+            OpenTelemetry.getTracerFactory().get(INSTRUMENTATION_LIB_ID).withSpan(parentSpan);
+        Scope contextScope =
+            OpenTelemetry.getDistributedContextManager().withContext(parentContext)) {
+      HttpRequestContext context = handler.handleStart(null, null, data, data);
+      Span currentSpan = handler.getSpanFromContext(context);
+      LOGGER.log(Level.INFO, currentSpan.getContext().toString());
+      try {
+        assertEquals(parentSpan.getContext().getTraceId(), currentSpan.getContext().getTraceId());
+        assertFalse(
+            parentSpan.getContext().getSpanId().equals(currentSpan.getContext().getSpanId()));
+        assertTrue(currentSpan.isRecording());
+      } finally {
+        handler.handleEnd(context, data, data, null);
+      }
+      assertFalse(inMemoryTracing.getFinishedSpanItems().isEmpty());
+      printSpans();
```

Comment:
unit tests shouldn't spam the logs. please remove these calls.

Reply:
Changed logging levels so does not print during normal builds. If developer wants to look at them, they can turn up the logging.